scripts/generateDeepExports.mjs

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env node
+// @ts-check
+/**
+ * Script to generate deep exports for pro and premium package versions.
+ *
+ * This script creates individual export files for each component/utility
+ * so they can be imported as:
+ * import { PieChart } from '@mui/x-charts-pro/PieChart'
+ * import { DataGrid } from '@mui/x-data-grid-premium/DataGrid'
+ */
+
+import fs from 'fs';
+import path from 'path';
+
+// Define package mappings (base -> pro])
+const PACKAGE_MAPPINGS = {
+  'x-charts': ['x-charts-pro', 'x-charts-premium'],
+  // 'x-data-grid': ['x-data-grid-pro', 'x-data-grid-premium'],
+  // 'x-date-pickers': ['x-date-pickers-pro'],
+  // 'x-tree-view': ['x-tree-view-pro'],
+};
+
+const IDENTIFIER_LINE =
+  '// Re-export automatically generated, to customize, simply remove this line.';
+
+/**
+ * Get component directories from src folder
+ * @param {string} packagePath
+ * @returns {string[]}
+ */
+function getComponentDirectories(packagePath) {
+  const srcPath = path.join(packagePath, 'src');
+
+  if (!fs.existsSync(srcPath)) {
+    return [];
+  }
+
+  return (
+    fs
+      .readdirSync(srcPath, { withFileTypes: true })
+      .filter((dirent) => dirent.isDirectory())
+      .map((dirent) => dirent.name)
+      .filter((name) => {
+        // Filter out utility directories that typically don't contain main components
+        const utilityDirs = [
+          'internals',
+          'context',
+          'hooks',
```

Comment:
Shouldn't this be configured by package? 

Also, shouldn't we remove `hooks` from this? 

For example, the `SeriesDemo` [uses deep imports](https://github.com/mui/mui-x/blob/a4c20f8d2ba2d374ef6c20e26b924945d70d5ac3/docs/data/charts/components/SeriesDemo.tsx#L5) from the `hooks` directory: `import { useLineSeries, useXAxis, useXScale, useYScale } from '@mui/x-charts/hooks';`

Reply:
🤔 I assumed we were already exporting hooks from `pro/hooks/index.ts` but we are not. However the file exists, so I would rather we just add it there.

This is a naive approach, so it still requires us to do some work to ensure everything is correct

Reply:
Yeah, this is already a step forward.

I wonder if we should have a auto generated section in a file instead. 

We can use something like:

```ts
// User code here

// # region auto-generated

// Generated code here

// # endregion

```

So we can gradually move the `hooks` export to auto generated ones. 

Something for the future, though.

Reply:
We could, but it also is a good recipe to create conflicts 🫠